pyvista/plotting/plotting.py

```diff
@@ -262,7 +258,7 @@ def __init__(
 
         # Add self to open plotters
         self._id_name = f"{hex(id(self))}-{len(_ALL_PLOTTERS)}"
-        _ALL_PLOTTERS[self._id_name] = self
+        _ALL_PLOTTERS[self._id_name] = weakref.ref(self)
```

Comment:
I had the same idea for https://github.com/pyvista/pyvista/pull/3037 but then I realised that our plot sphinx extension relies on this to pick out plotters from, and I'm pretty sure we can't let these figures be garbage collected.

If I'm right, we could make these weakrefs real refs for `BUILDING_GALLERY`... all in the name of practicality.

Reply:
It's quite likely this only worked because the plotter wasn't getting collected.

I think the approach of only tracking plotters when building the gallery is a better approach.